flocx_market/db/sqlalchemy/api.py

```diff
@@ -162,26 +162,49 @@ def contract_destroy(contract_id):
 
 
 # offer_contract_relationship
-def offer_contract_relationship_get(marketplace_offer_id=None,
-                                    contract_id=None):
-
-    if (contract_id is not None) and (marketplace_offer_id is None):
-        return get_session().query(models.OfferContractRelationship).filter_by(
-            contract_id=contract_id).all()
-
-    elif (contract_id is None) and (marketplace_offer_id is not None):
-        return get_session().query(models.OfferContractRelationship).filter_by(
-            marketplace_offer_id=marketplace_offer_id).all()
-    elif (contract_id is not None) and (marketplace_offer_id is not None):
-        return get_session().query(models.OfferContractRelationship) \
-            .filter(models.OfferContractRelationship.contract_id
-                    == contract_id and
-                    models.OfferContractRelationship.marketplace_offer_id
-                    == marketplace_offer_id).first()
-
-
-def offer_contract_relationship_get_all():
-    return get_session().query(models.OfferContractRelationship).all()
+def offer_contract_relationship_get(ocr_id=None):
+
+    if ocr_id is None:
+        return None
+    return get_session().query(models.OfferContractRelationship).filter(
+        models.OfferContractRelationship.offer_contract_relationship_id
+        == ocr_id).first()
+
+
+def offer_contract_relationship_get_all(filters=None):
+    if filters is None:
+        return get_session().query(models.OfferContractRelationship).all()
+    else:
+        query_params = {}
+        for field in ['marketplace_offer_id', 'contract_id', 'status']:
+            if field in filters:
+                query_params[field] = filters[field]
+
```

Comment:
Instead of the lines below, can't you just chain the query filters as in https://github.com/openstack/ironic/blob/master/ironic/db/sqlalchemy/api.py#L307-L308 ?

Reply:
it worked!